### ISSUE TITLE
f0 extraction with invDDSP or CREPE

### ISSUE THREAD

**Issue by adrienchaton** [magenta/ddsp#346]
Hi everyone,

I have a few questions regarding using either invDDSP or CREPE for extracting f0 envelopes to train a DDSP decoder.
Thanks for any thoughts or pointers on that, I am currently running a few tests so I can share on that too.
I already have my own DDSP decoders trained, next starting point is your collab notebook on pitch extraction with invDDSP.

1) I am running it locally, it appears that loading the pretrained invDDSP does not work with ddsp==1.0.1 .. are there some fixes I could do to the gin files to make them compatible please ?

2) I see you compare with CREPE using viterbi=False whereas default for training DDSP is viterbi=True, I am planning to benchmark these variations but wonder about your thoughts and motivations on viterbi=False.

3) although I trained my DDSP models with CREPE, my idea is to use a pretrained model on e.g. violin, collect a bunch of solo tracks which have nothing to do with URMP or any of the training data of either invDDSP or CREPE
then compute the reconstruction loss when feeding either CREPE or invDDSP and assuming that the largest factor of loss would be f0 error I could compare them

4) based on that I would retrain some models and maybe replace CREPE by invDDSP (hence I asked about compatibility with 1.0.1, it would be easier to have pitch extraction and decoder on the same version) .. have you tried that ? Could you get better models ?

5) edit: just noticing the frame rate of 31.25 .. any thoughts on whether it is sufficient to train DDSP if linearly interpolated to the original frame rate of 250Hz ?

Have a nice day

**Comment by jesseengel** [comment]
Sorry for the long delay

> I am running it locally, it appears that loading the pretrained invDDSP does not work with ddsp==1.0.1 .. are there some fixes I could do to the gin files to make them compatible please ?

Sorry, there were some breaking changes deeper in the code that we haven't have time to support. As of now, the ddsp-inv is more of a proof of principle experiment, so needs the fixed version to still work.

> I see you compare with CREPE using viterbi=False whereas default for training DDSP is viterbi=True, I am planning to benchmark these variations but wonder about your thoughts and motivations on viterbi=False.

This is just to keep the comparison fair, we could of applied viterbi decoding to ddsp-inv as well, but it was easier to just remove it from CREPE for the paper

> although I trained my DDSP models with CREPE, my idea is to use a pretrained model on e.g. violin, collect a bunch of solo tracks which have nothing to do with URMP or any of the training data of either invDDSP or CREPE
then compute the reconstruction loss when feeding either CREPE or invDDSP and assuming that the largest factor of loss would be f0 error I could compare them. based on that I would retrain some models and maybe replace CREPE by invDDSP (hence I asked about compatibility with 1.0.1, it would be easier to have pitch extraction and decoder on the same version) .. have you tried that ? Could you get better models ?

we've only really tried ddsp-inv on the datasets from the paper (urmp, mir-eval, mdb-stem-synth), but I imagine retraining CREPE or ddsp-inv on your specific data could perhaps help.

> edit: just noticing the frame rate of 31.25 .. any thoughts on whether it is sufficient to train DDSP if linearly interpolated to the original frame rate of 250Hz ?

my guess is that would be fine, we used a lower frame rate for computational savings if I remember correctly
